uwblueprint/planet-read
# Add database migrations to repository

## Notion ticket link

[Add database migrations to repository](https://www.notion.so/uwblueprintexecs/Add-database-migrations-to-repository-9dbf580ccd994b90aa613e97a26fc7f2)



## Implementation description
* Generate initial database migration using Flask-Migrate



## Steps to test
1. Add the seed_db.sh `AUTH_ID`s to your root `.env` file
2. Rebuild containers: `docker-compose down --volumes` and `docker-compose up --build`
3. Run `docker exec -it planet-read_py-backend_1 /bin/bash -c "flask db upgrade"`
4. Verify that the database tables are generated by checking the database itself
5. Run `docker exec -it planet-read_py-backend_1 /bin/bash -c "flask db downgrade"` and verify the tables are dropped: `docker exec -it planet-read_db_1 mysql -u root -proot -e "USE planet-read; SHOW TABLES;"` 


## What should reviewers focus on?
* `backend/python/migrations/versions/3f6fff20884c_initial_migration.py` (most of the other files in `backend/python/migrations/` were auto-generated/modified based on Starter Code)
~~* Is there a better place to put the seeded data than in the initial migration itself? This seemed to be the pattern that turned up the most when researching. A reference to the create table object is used in the `op.bulk_insert()` command, but an alternate command can be used if we want to separate it out (i.e. `op.execute()`. However, I believe alembic revisions are linear, so if we create a separate migration specifically for test data and then decide to remove it once we release to production, we will need to make sure we update the revision heads of the migrations to ensure the `flask db upgrade` command runs.~~
* The `AUTH_ID_[1-7]`s that were used for the seed_db.sh would need to be added to the vault

## Checklist
